### PR TITLE
Documentation - Refactor promotion banner without jQuery & add Wagtail Space 2024 banner

### PR DESCRIPTION
**Pull request by lb-**
## Overview

Refactors the current promotion banner without jQuery and sets up the banner for the 2024 Wagtail Space events. This includes a fix for cookie warnings and is a bit more configurable (with data attributes) for future banner updates.

Note: There are no exact dates for the US one yet so I have kept that intentionally generic.

## Screenshots

Screenshot 2024-01-31 at 7 58 36 am

Screenshot 2024-01-31 at 7 58 15 am

Screenshot 2024-01-31 at 7 58 07 am

**Validation of cookie (on clicking 'close') and refreshing.**

Screenshot 2024-01-31 at 8 01 55 am

## Details

- This approach will no longer work in IE11 but will work for all other modern browsers, even very old versions of them.
- Use HTML template in layout for content, setting data attributes to set expiry so that we do not have to update JS each time we do a new banner.
- Update CSS classes to be a bit more generic.
- Ensure that the cookie has the `sameSite` value set correctly to avoid browser console warnings and errors.
- Set closed cookie duration to 90 days (instead of 30) - this is better aligned to what users may expect when clearing the banner.
- Add upcoming Wagtail Space events 2024 & adjust styles to suit two links.
- Use non-jQuery JS. I have tried to keep the JS as close to ES5 as practical as this is for the documentation site.
- Update Eslint config as we no longer use jQuery in our documentation.
- Includes a slightly larger close icon to adhere to WCAG 2.2 minimum target button size.
- Avoids window namespace pollution by wrapping code in an IIFE
- Rework of a previous PR https://github.com/wagtail/wagtail/pull/8605
- Unblocks Sphinx Theme upgrade to Bootstrap 5 (without jQuery) https://github.com/wagtail/sphinx-wagtail-theme/issues/254

### CODE REVIEW

File: docs/_static/js/banner.js [hunk 1]
```diff
@@ -1,39 +1,69 @@
...
+    const dateUntil = new Date(template.dataset.dateUntil || 'INACTIVE');
+    if (!(dateUntil instanceof Date) || Number.isNaN(dateUntil.valueOf())) {
+      return;
+    }
+    // Pause any promotional banner for X days when manually cleared
+    const COOKIE_DURATION = Math.max(
+      Number(template.dataset.clearDuration || 30),
+      30,
+    );
+    const COOKIE_NAME = 'WAGTAIL_PROMOTION_BANNER_CLOSED';
```
Comment (allcaps) [inline docs/_static/js/banner.js:52]: The cookie name was unique per event to make sure new banners display regardless of the user closing the previous event banners.

Maybe we should add a comment to explain that?

**Review by allcaps — APPROVED** [review]
Awesome! Thank you.

Maybe we should advice unique cookie names per event?

But otherwise, LGTM!

File: docs/_static/js/banner.js (hunk 1 above)
Comment (lb-) [reply docs/_static/js/banner.js:52]: Good call. I can make this pull from a data attribute also so it's easier to update. 

Alternatively. I can append the expiry date to the cookie name. This way we get a unique Cooke name for any changes promotion expiry.

File: docs/_templates/layout.html [hunk 2]
```diff
@@ -89,3 +89,25 @@
...
+                    <strong>Wagtail Space NL</strong>
+                    <br>12 &ndash; 14 June 2024 in Arnhem, The Netherlands
+                </div>
+            </a>
+            <a href="https://us.wagtail.space/">
+                <div>
+                    <strong>Wagtail Space US</strong>
+                    <br>June 2024 in Philadelphia, USA
+                </div>
+            </a>
```
Comment (allcaps) [inline docs/_templates/layout.html:109]: I think most people see banners as one big clickable area. So, is it obvious enough that these are two sparate links? Do we need some visual distinction? Maybe a clear hover state is enough? 

And, does the US team want to advertise the event? Even if the date isn't final yet? Seems a bit strange to me to grab peoples attention with something that is not yet actionable.

Comment (lb-) [reply docs/_templates/layout.html:109]: They do have a hover state that's separate.

Also the Wagtail Space US link is live with a form to sign up for info about the 2024 event.

https://us.wagtail.space/

File: docs/_static/js/banner.js (hunk 1 above)
Comment (lb-) [reply docs/_static/js/banner.js]: Just updated this to have a more unique cookie name for each promotion based on the date the promotion is meant to run until.

@allcaps

File: docs/_templates/layout.html (hunk 2 above)
Comment (lb-) [reply docs/_templates/layout.html:109]: I also think that they are quite isolated visually and have clear headers.

Alternatively we could update back to one link with both bits of information and a link just to the main Wagtail space website.

Comment (allcaps) [reply docs/_templates/layout.html:109]: The original was better. If the links have hover states, that should suffice.

Cool! Thank you for the efforts!

**Comment by lb-** [comment]
Added to 6.0 branch a661e143d9085cc851fc40b3bc5304be25ad6b8d

**Comment by Stormheg** [comment]
Thanks all!

**Pull request merged**